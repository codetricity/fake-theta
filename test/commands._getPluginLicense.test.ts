import { request, toMatchMissingParameterError } from './helper';

describe('POST /osc/commands/execute _getPluginLicense', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getPluginLicense',
        parameters: {
          packageName: 'dummy',
        },
      })
      .expect(200);
    expect(res.text).toMatchSnapshot();
  });

  it('should respond missingParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({
        name: 'camera._getPluginLicense',
        parameters: { fileUri: 'dummy' },
      })
      .expect(400);
    toMatchMissingParameterError(res);
  });
});
