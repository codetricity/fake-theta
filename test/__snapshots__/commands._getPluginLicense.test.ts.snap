// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute _getPluginLicense should respond with results 1`] = `
"<html>
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=windows-1252" />
    <style type="text/css">
      body {
        padding: 0;
        font-family: sans-serif;
      }
    </style>
  </head>
  <body topmargin="0" leftmargin="0" rightmargin="0" bottommargin="0">
    <div>
      <p>This plug-in does not use open source software.</p>
    </div>
  </body>
</html>
"
`;
